src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/UuidIdentityProvider.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service;
+
+import java.util.UUID;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UuidIdentityProvider implements IdentityProvider {
+
+    public String identity() {
```

Comment:
I think we are duplicating logic. We can reuse HomeOfficeRequestUuidGenerator.generateUuid()

Reply:
I will get rid off the old implementation HomeOfficeRequestUuidGenerator in a separate cleanup task along with the INSTRUCT_STATUS flag. The above implementation (UuidIdentityProvider) was added as it gives us the ability to mock and verify for testing.